rainier-example/src/main/scala/com/stripe/rainier/example/stan/LowDimGaussMix.scala

```diff
@@ -0,0 +1,304 @@
+package com.stripe.rainier.example.stan
+
+//from the stan stat_comp_benchmarks
+
+import com.stripe.rainier.compute._
+import com.stripe.rainier.core._
+import com.stripe.rainier.sampler._
+import com.stripe.rainier.repl._
+
+object LowDimGaussMix {
+  def model(ys: Seq[Double]): RandomVariable[Map[String, Real]] =
+    for {
+      (mu1, sigma1) <- muSigma()
+      (mu2, sigma2) <- muSigma()
+      theta <- Beta(5, 5).param
+      _ <- Mixture(
+        Map(
+          Normal(mu1, sigma1) -> theta,
+          Normal(mu2, sigma2) -> (Real.one - theta)
+        )).fit(ys)
+    } yield
+      Map(
+        "mu[1]" -> mu1,
+        "mu[2]" -> mu2,
+        "sigma[1]" -> sigma1,
+        "sigma[2]" -> sigma2,
+        "theta" -> theta
+      )
+
+  def muSigma(): RandomVariable[(Real, Real)] =
+    for {
+      mu <- Normal(0, 2).param
+      sigma <- Normal(0, 2).param
+    } yield (mu, sigma.abs)
+
+  def main(args: Array[String]): Unit = {
+    val m = model(ys)
+    val t0 = System.nanoTime
+    val params = m.sample(HMC(5), warmups, samples, thin)
+    val t1 = System.nanoTime
+    val seconds = (t1 - t0) / 1e9
+    println(s"$samples samples in $seconds seconds")
+    List("mu[1]", "mu[2]", "sigma[1]", "sigma[2]", "theta").foreach { k =>
+      val mean = params.map { _(k) }.sum / params.size
+      println(s"$k = $mean")
+    }
+  }
+
+  val warmups = 1000
+  val samples = 10000
+  val thin = 10
+
+  val ys = List(-3.58542974746427, 3.11625192303286, 3.55429942030185,
```

Comment:
as a rule of thumb it's good to provide a type argument `Double` explicitly when a large number of elements are involved. It helps to drive type inference.